vividus-extension-selenium/src/main/java/org/vividus/selenium/AbstractVividusWebDriverFactory.java

```diff
@@ -54,13 +56,13 @@ public AbstractVividusWebDriverFactory(boolean remoteExecution, IWebDriverManage
     public VividusWebDriver create()
     {
         VividusWebDriver vividusWebDriver = new VividusWebDriver();
-        setDesiredCapabilities(vividusWebDriver.getDesiredCapabilities());
+        vividusWebDriver.setDesiredCapabilities(composeDesiredCapabilities(vividusWebDriver.getDesiredCapabilities()));
```

Comment:
it looks tricky: you get and set at the same time, what's the profit? why not to allow mutability?

Reply:
- VividusWebDriver itself is weird, we have empty desired capabilities as default value and at the same time we have a setter instead of having some methods to add values to the default desired capabilities
- we do not have a method like putAll in desired capabilities class, so to avoid foreach on out side I just create and set new one